UltimateMedals.as

```diff
@@ -1,55 +1,27 @@
-const array<string> medals = {
-	"\\$444" + Icons::Circle, // no medal
-	"\\$964" + Icons::Circle, // bronze medal
-	"\\$899" + Icons::Circle, // silver medal
-	"\\$db4" + Icons::Circle, // gold medal
-#if TMNEXT || MP4
-	"\\$071" + Icons::Circle, // author medal
-#elif TURBO
-	"\\$0f1" + Icons::Circle, // trackmaster medal
-	"\\$964" + Icons::Circle, // super bronze medal
-	"\\$899" + Icons::Circle, // super silver medal
-	"\\$db4" + Icons::Circle, // super gold medal
-	"\\$0ff" + Icons::Circle, // super trackmaster medal
-#endif
-};
+Medal@ g_personalBest = PersonalBestMedal();
 
-#if TMNEXT || MP4
-Record@ author = Record(authorText, 4, -5);
-#elif TURBO
-Record@ stmaster = Record(stmasterText, 8, -9);
-Record@ sgold = Record(sgoldText, 7, -8);
-Record@ ssilver = Record(ssilverText, 6, -7);
-Record@ sbronze = Record(sbronzeText, 5, -6);
-Record@ tmaster = Record(tmasterText, 4, -5);
+array<Medal@> g_medals = {
+#if TURBO
+	SuperTrackmasterMedal(), SuperGoldMedal(), SuperSilverMedal(), SuperBronzeMedal(),
 #endif
-Record@ gold = Record(goldText, 3, -4);
-Record@ silver = Record(silverText, 2, -3);
-Record@ bronze = Record(bronzeText, 1, -2);
-Record@ pbest = Record(pbestText, 0, -1, "\\$0ff");
-
-#if TMNEXT || MP4
-array<Record@> times = {author, gold, silver, bronze, pbest};
-#elif TURBO
-array<Record@> times = {stmaster, sgold, ssilver, sbronze, tmaster, gold, silver, bronze, pbest};
+	AuthorMedal(), GoldMedal(), SilverMedal(), BronzeMedal(), g_personalBest,
+};
 
-bool campaignMap = false;
+#if TURBO
+bool g_turboCampaignMap = false;
+int g_turboCampaignMapNumber = 0;
 #endif
 
-UI::Font@ font = null;
-
-uint64 limitMapNameLengthTime = 0;
-uint64 limitMapNameLengthTimeEnd = 0;
+UI::Font@ g_font = null;
 
+uint64 g_limitMapNameLengthTime = 0;
+uint64 g_limitMapNameLengthTimeEnd = 0;
 
-bool held = false;
 void OnKeyPress(bool down, VirtualKey key)
 {
-	if(key == windowVisibleKey && !held)
-	{
+	if (down && key == windowVisibleKey) {
 		windowVisible = !windowVisible;
 	}
-	held = down;
 }

```

Comment:
The purpose of `held` was to prevent flashing the window upon key repeat. Because I got multiple down-events (after the repeat delay) followed by a single up-event. Though maybe that's not really an issue.

Reply:
I figured; I don't think fixing that is worth the additional complexity though 😄